celery_batches/__init__.py

```diff
@@ -289,6 +289,21 @@ def task_message_handler(message, body, ack, reject, callbacks, **kw):
 
         return task_message_handler
 
+    def apply(self, args=None, kwargs=None, *_args, **_kwargs):
+        if self._get_app().conf.task_always_eager:
```

Comment:
When called from `task.apply_async()`, apparently `task.apply()` doesn't get a `self`??? (This is confirmed with a new test in celery-batches, which I haven't added to the PR yet as it is failing).

By this I mean that a call to task `add.defer(1)` will call through to `add.apply(args=(1,), kwargs={}, ...)`, but that will reach this line as `self=(1,), args=(), kwargs={}, ...`???

Reply:
I'm not sure why that would be happening, sorry. Seems like `apply` is unbound for some reason though...

Looks like the standard version of this does expect self: https://github.com/celery/celery/blob/cf829307991da3815e1f7b105e736d13dbc7a325/celery/app/task.py#L729-L732

Can you push the test so I can see the error stack?

Reply:
https://github.com/percipient/celery-batches/pull/16/commits/ebf9b20cccc225b73e292bf536e6b4d56a6b8922

Reply:
I have investigated and found that all methods on the task that are added or overwritten in the Batches class appear to be functions, rather than bound methods...  I have no idea what could be causing this...

Reply:
I've discovered the problem, will push another commit soon.

Reply:
Ah good find! Yeah I wish they would just get rid of the old style `Task` class. 👍 I don't believe there was any particular reason one was used over the other, besides that this was based off the old versions that were in the GitHub repo.